README.md

```diff
@@ -0,0 +1,109 @@
+# Parity SCALE Codec
+
+Rust implementation of the SCALE (Simple Concatenated Aggregate Little-Endian) data format for types used in the Parity Substrate framework.
+
+<!-- Inspired from Gav's overview written for Subtrate docs site -->
+SCALE is a light-weight format which allows encoding (and decoding) which makes it highly suitable for resource-constrained execution environments like blockchain runtimes and low-power, low-memory devices.
+
+It is important to note that the encoding context (knowledge of how the types and data structure looks like) needs to be known separately at both encoding and decoding ends. The encoded data does not include this contextual information.
+
+To get a better understanding of how the encoding is done for different types, take a look at the [low-level data formats overview page at the Substrate docs site](https://docs.substrate.dev/docs/low-level-data-formats).
+
+## Implementation
+
+The codec is implemented using the following traits:
+
+### Encode
+
+The `Encode` trait is used for encoding of data into the SCALE format. The `Encode` trait contains the following functions:
+
+* `size_hint(&self) -> usize`: Get the capacity (in bytes) required for the encoded data. This is needed to make sure we are not double-allocating memory needed for the encoding. This just needs to be an estimate and does not have to be the exact number. If you don't know the exact size, give the maximum.
```

Comment:
Hmm. It's correct, but should be changed. 
This is **required** to be a cheap operation, so should not involve iterations etc. 

Your sentence sounds so demanding, that the user needs to make sure that no double allocation is done.

And also if the size is not known, even no good maximum. Just don't implement it :)